kinder/ci/workflows/external-ca-tasks.yaml

```diff
@@ -0,0 +1,234 @@
+        docker cp $TMP_DIR/ca.key $CLUSTER-control-plane-2:/etc/kubernetes/pki/ca.key
+        docker cp $TMP_DIR/sa.key $CLUSTER-control-plane-2:/etc/kubernetes/pki/sa.key
+        docker cp $TMP_DIR/sa.pub $CLUSTER-control-plane-2:/etc/kubernetes/pki/sa.pub
+        docker cp $TMP_DIR/front-proxy-ca.crt $CLUSTER-control-plane-2:/etc/kubernetes/pki/front-proxy-ca.crt
+        docker cp $TMP_DIR/front-proxy-ca.key $CLUSTER-control-plane-2:/etc/kubernetes/pki/front-proxy-ca.key
+        docker cp $TMP_DIR/etcd/ca.key $CLUSTER-control-plane-2:/etc/kubernetes/pki/etcd/ca.key
+        docker cp $TMP_DIR/etcd/ca.crt $CLUSTER-control-plane-2:/etc/kubernetes/pki/etcd/ca.crt
+
+        # Copy shared kubeconfig files to the second CP Node
+        docker cp $TMP_DIR/admin.conf $CLUSTER-control-plane-2:/etc/kubernetes/admin.conf
+        docker cp $TMP_DIR/scheduler.conf $CLUSTER-control-plane-2:/etc/kubernetes/scheduler.conf
+        docker cp $TMP_DIR/controller-manager.conf $CLUSTER-control-plane-2:/etc/kubernetes/controller-manager.conf
+
+        # Copy the CA cert and key to the workers so that a kubeconfig.conf can be later generated
+        docker cp $TMP_DIR/ca.crt $CLUSTER-worker-1:/etc/kubernetes/pki/ca.crt
+        docker cp $TMP_DIR/ca.key $CLUSTER-worker-1:/etc/kubernetes/pki/ca.key
+        docker cp $TMP_DIR/ca.crt $CLUSTER-worker-2:/etc/kubernetes/pki/ca.crt
+        docker cp $TMP_DIR/ca.key $CLUSTER-worker-2:/etc/kubernetes/pki/ca.key
+
+        # Remove the temporary directory
+        rm -rf ${TMP_DIR}
+
+        # Generate remaining certificates and kubelet.conf for bootstrap.
+        # Using valid embedded credentials in a kubelet.conf makes the kubelet bootstrap this Node right away,
+        # without depending on the KCM to sign the rotatable Node client credentials. Normally users should either
+        # disable the kubelet client certificate rotation and/or rely on an external signer.
+        docker exec $CLUSTER-control-plane-2 kubeadm init phase certs all --control-plane-endpoint=$CPE --v=$VERBOSITY
+        docker exec $CLUSTER-control-plane-2 kubeadm init phase kubeconfig kubelet --control-plane-endpoint=$CPE --v=$VERBOSITY
+
+        # Generate a kubelet.conf on the workers for bootstrap
+        docker exec $CLUSTER-worker-1 kubeadm init phase kubeconfig kubelet --control-plane-endpoint=$CPE --v=$VERBOSITY
+        docker exec $CLUSTER-worker-2 kubeadm init phase kubeconfig kubelet --control-plane-endpoint=$CPE --v=$VERBOSITY
+
+        # Remove the ca.key from all Nodes.
+        # The KCM tolerates the --cluster-signing-key flag, pointing to a file that is missing.
+        docker exec $CLUSTER-control-plane-1 rm /etc/kubernetes/pki/ca.key
+        docker exec $CLUSTER-control-plane-2 rm /etc/kubernetes/pki/ca.key
+        docker exec $CLUSTER-worker-1 rm /etc/kubernetes/pki/ca.key
+        docker exec $CLUSTER-worker-2 rm /etc/kubernetes/pki/ca.key
```

Comment:
[1] Long term I would like this to became `kinder do setup-ca`, so we the logic is in go making it easier to test a similar setup locally. Do you mind to open an issue for this?
@yagonobre ^^

Reply:
I can log an issue for that.

Reply:
updated the PR to include the logic for the new suggested action.